jsonpatch.py

```diff
@@ -817,17 +817,17 @@ def _compare_lists(self, path, src, dst):
                 self._item_added(path, key, dst[key])
 
     def _compare_values(self, path, key, src, dst):
-        if src == dst:
-            return
-
-        elif isinstance(src, MutableMapping) and \
+        if isinstance(src, MutableMapping) and \
                 isinstance(dst, MutableMapping):
             self._compare_dicts(_path_join(path, key), src, dst)
 
         elif isinstance(src, MutableSequence) and \
                 isinstance(dst, MutableSequence):
             self._compare_lists(_path_join(path, key), src, dst)
 
+        elif json.dumps(src) == json.dumps(dst):
```

Comment:
Could you please ask a comment to describe this comparison? It makes sense now in the context of this pull request, but it might be confusing if you look at it a year from now.

Reply:
Done, plus a note about how performance could later be improved.